### PR TITLE
influxdb2: 2.0.8 -> 2.1.1

### PR DESCRIPTION
**Pull request by danderson**
* libflux: 0.124.0 -> 0.139.0; as specified in influxdb 2.1's release notes

* UI: 2.0.8 -> 2.1.2; matching what influxdb 2.1.1 requires

* Add checks in influxdb2's build to verify that libflux and the UI are at
  the version specified by upstream.

* The CLI got split into a separate repository for reasons to do with influx's
  cloud service, and its version has also been decoupled from the main influxdb
  repository. Split the derivation into two, and add a `buildEnv` derivation for
  temporary backwards compatibility.

###### Motivation for this change

New upstream release. Unfortunately not an easy upgrade, various structural changes to the source and build system had to be adjusted for.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by avdv — APPROVED** [review]
:+1: nice!

**Comment by danderson** [comment]
@SuperSandro2000 can you take a look? avdv approved, but it needs merging if we're happy.

**Review by abbradar — CHANGES_REQUESTED** [review]
We need to drop our UI stuff now that it's prebuilt (see https://github.com/NixOS/nixpkgs/pull/150815/files).

To discuss: I'd say that it's better to have separate packages for ui and for the server, and one package that does `buildEnv` for backwards compatibility (which we can drop later). This both makes it more parallel and gives us ability to bump them separately.

**Comment by avdv** [comment]
> We need to drop our UI stuff now that it's prebuilt (see [#150815 (files)](https://github.com/NixOS/nixpkgs/pull/150815/files)).

What do you mean? The ui is no longer built since commit b23d8aa39cd2f49ec908d395adcbe7a70274ad9b already.

> To discuss: I'd say that it's better to have separate packages for ui and for the server, and one package that does `buildEnv` for backwards compatibility (which we can drop later). This both makes it more parallel and gives us ability to bump them separately.

I also would be in favor of splitting out the cli into its own package. (this is what you meant, right?)

**Comment by abbradar** [comment]
> What do you mean? The ui is no longer built since commit b23d8aa already.

We have several files [left over](https://github.com/NixOS/nixpkgs/tree/master/pkgs/servers/nosql/influxdb2) from that time along with update script that are no longer needed.

**Comment by avdv** [comment]
> We have several files [left over](https://github.com/NixOS/nixpkgs/tree/master/pkgs/servers/nosql/influxdb2?rgh-link-date=2021-12-16T08%3A54%3A07Z) from that time along with update script that are no longer needed.

Ouch, my bad.. Yes, these should be removed.

**Comment by danderson** [comment]
Thanks for the feedback! Splitting up sounds good, I'll assemble the changes now.

**Comment by danderson** [comment]
Changes made. @abbradar I added you as maintainer for both the CLI and server packages of influxdb2, is that okay?

File: pkgs/top-level/all-packages.nix
Comment (abbradar) [inline pkgs/top-level/all-packages.nix:21341]: Two small points:

* We should also use `influxdb2-server` in `influxdb2` NixOS module;
* Maybe add an explicit comment here stating that this is for backwards compatibility? I expect us to drop it in some release.

**Comment by abbradar** [comment]
@danderson Yes, thanks!

File: pkgs/top-level/all-packages.nix
Comment (danderson) [reply pkgs/top-level/all-packages.nix:21341]: Both done. I also added the package split as a "notable change" in the 22.05 release notes.

**Comment by danderson** [comment]
@abbradar done, and also added a release note about the package split. Looks good?

**Review by abbradar — APPROVED** [review]

**Comment by abbradar** [comment]
Let's wait till it builds and merge :)

**Pull request merged**
